### PR TITLE
Visual studio coverage fix null ref exception

### PR DESCRIPTION
**Pull request by BobSilent**
fix nullref exception for VisualStudio Coverage, where a complete file is not covered by tests:
CxxCoverageSensor
entry.getValue().createMeasures()
entry.getValue() would be be null.

VisualStudioParser:
therfore ignore files which are not in coverageData.

CoverageMeasuresBuilder builder = coverageData.remove(id);
would return null if id does not exists in coverageData

### CODE REVIEW

File: sonar-cxx-plugin/src/main/java/org/sonar/plugins/cxx/coverage/CxxCoverageSensor.java [hunk 1]
```diff
@@ -165,7 +165,9 @@ private void saveMeasures(SensorContext context,
       String filePath = entry.getKey();
       InputFile cxxFile = fs.inputFile(fs.predicates().hasPath(filePath));
       if (cxxFile != null) {
-        Collection<Measure> measures = entry.getValue().createMeasures();
+        CxxUtils.LOG.debug("Saving for file '{}'", filePath);
```
Comment (jmecosta) [inline sonar-cxx-plugin/src/main/java/org/sonar/plugins/cxx/coverage/CxxCoverageSensor.java]: does it need this additional debug information? there is another 2 lines below

File: sonar-cxx-plugin/src/main/java/org/sonar/plugins/cxx/coverage/CxxCoverageSensor.java [hunk 2]
```diff
@@ -165,7 +165,9 @@ private void saveMeasures(SensorContext context,
...
+        CoverageMeasuresBuilder builder = entry.getValue();
```
Comment (jmecosta) [inline sonar-cxx-plugin/src/main/java/org/sonar/plugins/cxx/coverage/CxxCoverageSensor.java]: why do we need to introduce this here?

Comment (BobSilent) [inline sonar-cxx-plugin/src/main/java/org/sonar/plugins/cxx/coverage/CxxCoverageSensor.java]: I had the case where an exception was thrown 
entry.getValue().createMeasures();
and it is easier to analyze if there is no chained method call

File: sonar-cxx-plugin/src/main/java/org/sonar/plugins/cxx/coverage/CxxCoverageSensor.java (hunk 1 above)
Comment (BobSilent) [inline sonar-cxx-plugin/src/main/java/org/sonar/plugins/cxx/coverage/CxxCoverageSensor.java]: it is in connection with the change below, an exception was thrown and i did not had any file information, therefore i added this line to debug log

File: sonar-cxx-plugin/src/main/java/org/sonar/plugins/cxx/coverage/CxxCoverageSensor.java (hunk 2 above)
Comment (jmecosta) [inline sonar-cxx-plugin/src/main/java/org/sonar/plugins/cxx/coverage/CxxCoverageSensor.java]: Sure but can't we print it in the exception handler. Now for nominal cases logs are going to be really big. We have done some changes to the plugin to keeps things to a minimal and this seems going the other team around

**Comment by guwirth** [comment]
@BobSilent please rebase and upload again. There a bigger changes in the plugin to be compatible with SQ 5.6. I close this one.

**Pull request closed without merging**